nautobot/core/views/mixins.py

```diff
@@ -120,13 +120,20 @@ def has_permission(self):
 
         return False
 
-    def check_permissions(self, request):
+    def dispatch(self, request, *args, **kwargs):
         """
+        Override the default dispatch() method to check permissions first.
         Used to determine whether the user has permissions to a view and object-level permissions.
         Using AccessMixin handle_no_permission() to deal with Object-Level permissions and API-Level permissions in one pass.
         """
-        if not self.has_permission():
-            self.handle_no_permission()
+        # initialize_request also instantiates self.action which is needed for permission checks.
+        self.initialize_request(request, *args, **kwargs)
+        has_permission = self.check_permissions(request)
+
+        if not has_permission:
```

Comment:
Was this not possible with the previous method override for `check_permission()` but just fixing it to `return` the output of `self.handle_no_permission()` instead of just calling it?

I'm asking, because I'd like to see us avoid having to overload `dispatch()` if we can!

Reply:
That was my initial approach but [this](https://github.com/encode/django-rest-framework/blob/56946fac8f29aa44ce84391f138d63c4c8a2a285/rest_framework/views.py#L415) is where drf calls self.check_permissions(). Unless we override initial() and do return self.check_permissions, self.handle_no_permission() (which is a redirect) does not do anything.

Reply:
Lemme know if im making sense whatsoever

Reply:
Ah. Got it. Check permissions isn't meant to return anything.